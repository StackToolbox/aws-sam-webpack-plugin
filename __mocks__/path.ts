const path = jest.genMockFromModule("path") as any;

interface PathMap {
  [fname: string]: string;
}

let mockBasename: PathMap = {};
path.__setMockBasenames = (basenames: PathMap) => {
  mockBasename = basenames;
};

let mockDirname: PathMap = {};
path.__setMockDirnames = (dirnames: PathMap) => {
  mockDirname = dirnames;
};

let mockRelative: PathMap = {};
path.__setMockRelatives = (relatives: PathMap) => {
  mockRelative = relatives;
};

path.__clearMocks = (): void => {
  mockBasename = {};
  mockDirname = {};
  mockRelative = {};
};

path.basename = (name: string): string => {
  if (name in mockBasename) {
    return mockBasename[name];
  }
  throw new Error(`Unknown file ${name}`);
};

path.dirname = (name: string): string => {
  if (name in mockDirname) {
    return mockDirname[name];
  }
  throw new Error(`Unknown dirname ${name}`);
};

path.relative = (folder: string, name: string): string => {
  if (`${folder}#${name}` in mockRelative) {
    return mockRelative[`${folder}#${name}`];
  }
  throw new Error(`Unknown relative ${folder}, ${name}`);
};

path.resolve = (folder: string): string => {
  return process.cwd() + "/" + folder;
};

module.exports = path;
