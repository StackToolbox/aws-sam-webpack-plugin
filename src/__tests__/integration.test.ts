import SamPlugin from "../index";
import fs from "fs";
import path from "path";

jest.mock("fs");
jest.mock("path");

const samTemplate = `
AWSTemplateFormatVersion: "2010-09-09"
Transform: AWS::Serverless-2016-10-31

Globals:
  Function:
    Runtime: nodejs10.x

Resources:
  MyLambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: src/my-lambda
      Handler: app.handler
`;

test("Happy path with default constructor works", () => {
  const plugin = new SamPlugin();

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  fs.__setMockDirs(["."]);
  // @ts-ignore
  fs.__setMockFiles({ "./template.yaml": samTemplate });

  // @ts-ignore
  path.__clearMocks();
  // @ts-ignore
  path.__setMockBasenames({ "./template.yaml": "template.yaml" });
  // @ts-ignore
  path.__setMockDirnames({ "./template.yaml": "." });
  // @ts-ignore
  path.__setMockRelatives({ ".#.": "" });

  const entryPoints = plugin.entry();

  let afterEmit: (_compilation: any) => void;

  plugin.apply({
    hooks: {
      afterEmit: {
        tap: (n: string, f: (_compilation: any) => void) => {
          afterEmit = f;
        },
      },
    },
  });
  // @ts-ignore
  afterEmit(null);

  // @ts-ignore
  expect({ entryPoints, files: fs.__getMockWrittenFiles() }).toMatchSnapshot();
});

test("Happy path with empty options in the constructor works", () => {
  const plugin = new SamPlugin({});

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  fs.__setMockDirs(["."]);
  // @ts-ignore
  fs.__setMockFiles({ "./template.yaml": samTemplate });

  // @ts-ignore
  path.__clearMocks();
  // @ts-ignore
  path.__setMockBasenames({ "./template.yaml": "template.yaml" });
  // @ts-ignore
  path.__setMockDirnames({ "./template.yaml": "." });
  // @ts-ignore
  path.__setMockRelatives({ ".#.": "" });

  const entryPoints = plugin.entry();

  let afterEmit: (_compilation: any) => void;

  plugin.apply({
    hooks: {
      afterEmit: {
        tap: (n: string, f: (_compilation: any) => void) => {
          afterEmit = f;
        },
      },
    },
  });
  // @ts-ignore
  afterEmit(null);

  // Does create a .vscode folder
  // @ts-ignore
  expect(fs.__getMockMakedirs().includes(".vscode")).toBeTruthy();

  // @ts-ignore
  expect({ entryPoints, files: fs.__getMockWrittenFiles() }).toMatchSnapshot();
});

test("Happy path with empty options in the constructor works and an existing .vscode folder", () => {
  const plugin = new SamPlugin({});

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  fs.__setMockDirs([".", ".vscode"]);
  // @ts-ignore
  fs.__setMockFiles({ "./template.yaml": samTemplate });

  // @ts-ignore
  path.__clearMocks();
  // @ts-ignore
  path.__setMockBasenames({ "./template.yaml": "template.yaml" });
  // @ts-ignore
  path.__setMockDirnames({ "./template.yaml": "." });
  // @ts-ignore
  path.__setMockRelatives({ ".#.": "" });

  const entryPoints = plugin.entry();

  let afterEmit: (_compilation: any) => void;

  plugin.apply({
    hooks: {
      afterEmit: {
        tap: (n: string, f: (_compilation: any) => void) => {
          afterEmit = f;
        },
      },
    },
  });
  // @ts-ignore
  afterEmit(null);

  // Does not create a .vscode folder
  // @ts-ignore
  expect(fs.__getMockMakedirs().includes(".vscode")).toBeFalsy();

  // @ts-ignore
  expect({ entryPoints, files: fs.__getMockWrittenFiles() }).toMatchSnapshot();
});

test("Happy path with VS Code debugging disabled", () => {
  const plugin = new SamPlugin({ vscodeDebug: false });

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  fs.__setMockDirs(["."]);
  // @ts-ignore
  fs.__setMockFiles({ "./template.yaml": samTemplate });

  // @ts-ignore
  path.__clearMocks();
  // @ts-ignore
  path.__setMockBasenames({ "./template.yaml": "template.yaml" });
  // @ts-ignore
  path.__setMockDirnames({ "./template.yaml": "." });
  // @ts-ignore
  path.__setMockRelatives({ ".#.": "" });

  const entryPoints = plugin.entry();

  let afterEmit: (_compilation: any) => void;

  plugin.apply({
    hooks: {
      afterEmit: {
        tap: (n: string, f: (_compilation: any) => void) => {
          afterEmit = f;
        },
      },
    },
  });
  // @ts-ignore
  afterEmit(null);

  // Does not create a .vscode folder
  // @ts-ignore
  expect(fs.__getMockMakedirs().includes(".vscode")).toBeFalsy();

  // @ts-ignore
  expect({ entryPoints, files: fs.__getMockWrittenFiles() }).toMatchSnapshot();
});

const vscodeLaunchJson1 = `{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "CustomLambda",
      "type": "node",
      "request": "attach",
      "address": "localhost",
      "port": 5858,
      "localRoot": "\${workspaceFolder}/.aws-sam/build/CustomLambda",
      "remoteRoot": "/var/task",
      "protocol": "inspector",
      "stopOnEntry": false,
      "outFiles": [
        "\${workspaceFolder}/.aws-sam/build/CustomLambda/**/*.js"
      ],
      "sourceMaps": true,
      "skipFiles": [
        "/var/runtime/**/*.js",
        "<node_internals>/**/*.js"
      ]
    }
  ]
}`;
const vscodeLaunchJson2 = `{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "CustomLambda",
      "type": "node",
      "request": "attach",
      "address": "localhost",
      "port": 5858,
      "localRoot": "\${workspaceFolder}/.aws-sam/build/CustomLambda",
      "remoteRoot": "/var/task",
      "protocol": "inspector",
      "stopOnEntry": false,
      "outFiles": [
        "\${workspaceFolder}/.aws-sam/build/CustomLambda/**/*.js"
      ],
      "sourceMaps": true,
      "skipFiles": [
        "/var/runtime/**/*.js",
        "<node_internals>/**/*.js"
      ]
    },
    // BEGIN AwsSamPlugin
    {
      "name": "OldLambda",
      "type": "node",
      "request": "attach",
      "address": "localhost",
      "port": 5858,
      "localRoot": "\${workspaceFolder}/.aws-sam/build/OldLambda",
      "remoteRoot": "/var/task",
      "protocol": "inspector",
      "stopOnEntry": false,
      "outFiles": [
        "\${workspaceFolder}/.aws-sam/build/OldLambda/**/*.js"
      ],
      "sourceMaps": true,
      "skipFiles": [
        "/var/runtime/**/*.js",
        "<node_internals>/**/*.js"
      ]
    }
    // END AwsSamPlugin
  ]
}`;
const vscodeLaunchJsonTest = `{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "CustomLambda",
      "type": "node",
      "request": "attach",
      "address": "localhost",
      "port": 5858,
      "localRoot": "\${workspaceFolder}/.aws-sam/build/CustomLambda",
      "remoteRoot": "/var/task",
      "protocol": "inspector",
      "stopOnEntry": false,
      "outFiles": [
        "\${workspaceFolder}/.aws-sam/build/CustomLambda/**/*.js"
      ],
      "sourceMaps": true,
      "skipFiles": [
        "/var/runtime/**/*.js",
        "<node_internals>/**/*.js"
      ]
    },
    // BEGIN AwsSamPlugin
    {
      "name": "MyLambda",
      "type": "node",
      "request": "attach",
      "address": "localhost",
      "port": 5858,
      "localRoot": "\${workspaceFolder}/.aws-sam/build/MyLambda",
      "remoteRoot": "/var/task",
      "protocol": "inspector",
      "stopOnEntry": false,
      "outFiles": [
        "\${workspaceFolder}/.aws-sam/build/MyLambda/**/*.js"
      ],
      "sourceMaps": true,
      "skipFiles": [
        "/var/runtime/**/*.js",
        "<node_internals>/**/*.js"
      ]
    }
    // END AwsSamPlugin
  ]
}`;

test.each([
  [vscodeLaunchJson1, vscodeLaunchJsonTest],
  [vscodeLaunchJson2, vscodeLaunchJsonTest],
])("Happy build launch.json with replace old content", (srcData, testData) => {
  const plugin = new SamPlugin({ vscodeDebug: true });

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  fs.__setMockDirs(["."]);
  // @ts-ignore
  fs.__setMockFiles({ "./template.yaml": samTemplate, ".vscode/launch.json": srcData });

  // @ts-ignore
  path.__clearMocks();
  // @ts-ignore
  path.__setMockBasenames({ "./template.yaml": "template.yaml" });
  // @ts-ignore
  path.__setMockDirnames({ "./template.yaml": "." });
  // @ts-ignore
  path.__setMockRelatives({ ".#.": "" });

  const entryPoints = plugin.entry();

  let afterEmit: (_compilation: any) => void;

  plugin.apply({
    hooks: {
      afterEmit: {
        tap: (n: string, f: (_compilation: any) => void) => {
          afterEmit = f;
        },
      },
    },
  });
  // @ts-ignore
  afterEmit(null);

  // @ts-ignore
  const vscodeLaunchJsonContent = fs.__getMockWrittenFiles()[".vscode/launch.json"];

  expect(vscodeLaunchJsonContent).toEqual(testData);
});

test("Happy path with multiple projects works", () => {
  const plugin = new SamPlugin({ projects: { a: "project-a", b: "project-b" } });

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  fs.__setMockDirs(["project-a", "project-b"]);
  // @ts-ignore
  fs.__setMockFiles({ "project-a/template.yaml": samTemplate, "project-b/template.yaml": samTemplate });

  // @ts-ignore
  path.__clearMocks();
  // @ts-ignore
  path.__setMockBasenames({ "project-a/template.yaml": "template.yaml", "project-b/template.yaml": "template.yaml" });
  // @ts-ignore
  path.__setMockDirnames({ "project-a/template.yaml": "project-a", "project-b/template.yaml": "project-b" });
  // @ts-ignore
  path.__setMockRelatives({ ".#project-a": "project-a", ".#project-b": "project-b" });

  const entryPoints = plugin.entry();

  let afterEmit: (_compilation: any) => void;

  plugin.apply({
    hooks: {
      afterEmit: {
        tap: (n: string, f: (_compilation: any) => void) => {
          afterEmit = f;
        },
      },
    },
  });
  // @ts-ignore
  afterEmit(null);

  // @ts-ignore
  expect({ entryPoints, files: fs.__getMockWrittenFiles() }).toMatchSnapshot();
});

test("Happy path with multiple projects and different template names works", () => {
  const plugin = new SamPlugin({ projects: { a: "project-a/template-a.yaml", b: "project-b/template-b.yaml" } });

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  fs.__setMockDirs(["project-a", "project-b"]);
  // @ts-ignore
  fs.__setMockFiles({ "project-a/template-a.yaml": samTemplate, "project-b/template-b.yaml": samTemplate });

  // @ts-ignore
  path.__clearMocks();
  // @ts-ignore
  path.__setMockBasenames({
    "project-a/template-a.yaml": "template-a.yaml",
    "project-b/template-b.yaml": "template-b.yaml",
  });
  // @ts-ignore
  path.__setMockDirnames({ "project-a/template-a.yaml": "project-a", "project-b/template-b.yaml": "project-b" });
  // @ts-ignore
  path.__setMockRelatives({ ".#project-a": "project-a", ".#project-b": "project-b" });

  const entryPoints = plugin.entry();

  let afterEmit: (_compilation: any) => void;

  plugin.apply({
    hooks: {
      afterEmit: {
        tap: (n: string, f: (_compilation: any) => void) => {
          afterEmit = f;
        },
      },
    },
  });
  // @ts-ignore
  afterEmit(null);

  // @ts-ignore
  expect({ entryPoints, files: fs.__getMockWrittenFiles() }).toMatchSnapshot();
});

test("Calling apply() before entry() throws an error", () => {
  const plugin = new SamPlugin();

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  path.__clearMocks();

  let afterEmit: (_compilation: any) => void;

  plugin.apply({
    hooks: {
      afterEmit: {
        tap: (n: string, f: (_compilation: any) => void) => {
          afterEmit = f;
        },
      },
    },
  });
  // @ts-ignore
  expect(() => afterEmit(null)).toThrowError("It looks like AwsSamPlugin.entry() was not called");
});

test("Happy path for filename() when the Lambda is found", () => {
  const plugin = new SamPlugin();

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  fs.__setMockDirs(["."]);
  // @ts-ignore
  fs.__setMockFiles({ "./template.yaml": samTemplate });

  // @ts-ignore
  path.__clearMocks();
  // @ts-ignore
  path.__setMockBasenames({ "./template.yaml": "template.yaml" });
  // @ts-ignore
  path.__setMockDirnames({ "./template.yaml": "." });
  // @ts-ignore
  path.__setMockRelatives({ ".#.": "" });

  const entryPoints = plugin.entry();

  expect(plugin.filename({ chunk: { name: "MyLambda" } })).toEqual("./.aws-sam/build/MyLambda/app.js");
});

test("Fails when filename() is passed an invalid lambda name", () => {
  const plugin = new SamPlugin();

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  fs.__setMockDirs(["."]);
  // @ts-ignore
  fs.__setMockFiles({ "./template.yaml": samTemplate });

  // @ts-ignore
  path.__clearMocks();
  // @ts-ignore
  path.__setMockBasenames({ "./template.yaml": "template.yaml" });
  // @ts-ignore
  path.__setMockDirnames({ "./template.yaml": "." });
  // @ts-ignore
  path.__setMockRelatives({ ".#.": "" });

  const entryPoints = plugin.entry();

  //console.log("XXX", plugin.filename({chunk: { name: "FakeLambda" }}));
  expect(() => plugin.filename({ chunk: { name: "FakeLambda" } })).toThrowError(
    "Unable to find filename for FakeLambda"
  );
});

test("Fails when there is no template.yaml or template.yml and you provided a directory", () => {
  const plugin = new SamPlugin();

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  fs.__setMockDirs(["."]);
  // // @ts-ignore
  // fs.__setMockFiles({ "./template.yaml": samTemplate });

  // @ts-ignore
  path.__clearMocks();
  // @ts-ignore
  // path.__setMockBasenames({ "./template.yaml": "template.yaml" });
  // // @ts-ignore
  // path.__setMockDirnames({ "./template.yaml": "." });
  // @ts-ignore
  path.__setMockRelatives({ ".#.": "" });

  expect(() => plugin.entry()).toThrowError("Could not find template.yaml or template.yml in .");
});

test("Happy path with an output file specified", () => {
  const plugin = new SamPlugin({ outFile: "index" });

  // @ts-ignore
  fs.__clearMocks();
  // @ts-ignore
  fs.__setMockDirs(["."]);
  // @ts-ignore
  fs.__setMockFiles({ "./template.yaml": samTemplate });

  // @ts-ignore
  path.__clearMocks();
  // @ts-ignore
  path.__setMockBasenames({ "./template.yaml": "template.yaml" });
  // @ts-ignore
  path.__setMockDirnames({ "./template.yaml": "." });
  // @ts-ignore
  path.__setMockRelatives({ ".#.": "" });

  const entryPoints = plugin.entry();

  let afterEmit: (_compilation: any) => void;

  plugin.apply({
    hooks: {
      afterEmit: {
        tap: (n: string, f: (_compilation: any) => void) => {
          afterEmit = f;
        },
      },
    },
  });
  // @ts-ignore
  afterEmit(null);

  // @ts-ignore
  expect({ entryPoints, files: fs.__getMockWrittenFiles() }).toMatchSnapshot();
});
