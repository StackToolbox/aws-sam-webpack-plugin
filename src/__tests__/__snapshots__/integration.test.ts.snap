// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Happy path with VS Code debugging disabled 1`] = `
Object {
  "entryPoints": Object {
    "MyLambda": "./src/my-lambda/app",
  },
  "files": Object {
    ".aws-sam/build/template.yaml": "AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Runtime: nodejs20.x
Resources:
  MyLambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: MyLambda
      Handler: app.handler
",
  },
}
`;

exports[`Happy path with an output file specified 1`] = `
Object {
  "entryPoints": Object {
    "MyLambda": "./src/my-lambda/app",
  },
  "files": Object {
    ".aws-sam/build/template.yaml": "AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Runtime: nodejs20.x
Resources:
  MyLambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: MyLambda
      Handler: index.handler
",
    ".vscode/launch.json": "{
  \\"version\\": \\"0.2.0\\",
  \\"configurations\\": [
    // BEGIN AwsSamPlugin
    {
      \\"name\\": \\"MyLambda\\",
      \\"type\\": \\"node\\",
      \\"request\\": \\"attach\\",
      \\"address\\": \\"localhost\\",
      \\"port\\": 5858,
      \\"localRoot\\": \\"\${workspaceFolder}/.aws-sam/build/MyLambda\\",
      \\"remoteRoot\\": \\"/var/task\\",
      \\"protocol\\": \\"inspector\\",
      \\"stopOnEntry\\": false,
      \\"outFiles\\": [
        \\"\${workspaceFolder}/.aws-sam/build/MyLambda/**/*.js\\"
      ],
      \\"sourceMaps\\": true,
      \\"skipFiles\\": [
        \\"/var/runtime/**/*.js\\",
        \\"<node_internals>/**/*.js\\"
      ]
    }
    // END AwsSamPlugin
  ]
}",
  },
}
`;

exports[`Happy path with default constructor works 1`] = `
Object {
  "entryPoints": Object {
    "MyLambda": "./src/my-lambda/app",
  },
  "files": Object {
    ".aws-sam/build/template.yaml": "AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Runtime: nodejs20.x
Resources:
  MyLambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: MyLambda
      Handler: app.handler
",
    ".vscode/launch.json": "{
  \\"version\\": \\"0.2.0\\",
  \\"configurations\\": [
    // BEGIN AwsSamPlugin
    {
      \\"name\\": \\"MyLambda\\",
      \\"type\\": \\"node\\",
      \\"request\\": \\"attach\\",
      \\"address\\": \\"localhost\\",
      \\"port\\": 5858,
      \\"localRoot\\": \\"\${workspaceFolder}/.aws-sam/build/MyLambda\\",
      \\"remoteRoot\\": \\"/var/task\\",
      \\"protocol\\": \\"inspector\\",
      \\"stopOnEntry\\": false,
      \\"outFiles\\": [
        \\"\${workspaceFolder}/.aws-sam/build/MyLambda/**/*.js\\"
      ],
      \\"sourceMaps\\": true,
      \\"skipFiles\\": [
        \\"/var/runtime/**/*.js\\",
        \\"<node_internals>/**/*.js\\"
      ]
    }
    // END AwsSamPlugin
  ]
}",
  },
}
`;

exports[`Happy path with empty options in the constructor works 1`] = `
Object {
  "entryPoints": Object {
    "MyLambda": "./src/my-lambda/app",
  },
  "files": Object {
    ".aws-sam/build/template.yaml": "AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Runtime: nodejs20.x
Resources:
  MyLambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: MyLambda
      Handler: app.handler
",
    ".vscode/launch.json": "{
  \\"version\\": \\"0.2.0\\",
  \\"configurations\\": [
    // BEGIN AwsSamPlugin
    {
      \\"name\\": \\"MyLambda\\",
      \\"type\\": \\"node\\",
      \\"request\\": \\"attach\\",
      \\"address\\": \\"localhost\\",
      \\"port\\": 5858,
      \\"localRoot\\": \\"\${workspaceFolder}/.aws-sam/build/MyLambda\\",
      \\"remoteRoot\\": \\"/var/task\\",
      \\"protocol\\": \\"inspector\\",
      \\"stopOnEntry\\": false,
      \\"outFiles\\": [
        \\"\${workspaceFolder}/.aws-sam/build/MyLambda/**/*.js\\"
      ],
      \\"sourceMaps\\": true,
      \\"skipFiles\\": [
        \\"/var/runtime/**/*.js\\",
        \\"<node_internals>/**/*.js\\"
      ]
    }
    // END AwsSamPlugin
  ]
}",
  },
}
`;

exports[`Happy path with empty options in the constructor works and an existing .vscode folder 1`] = `
Object {
  "entryPoints": Object {
    "MyLambda": "./src/my-lambda/app",
  },
  "files": Object {
    ".aws-sam/build/template.yaml": "AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Runtime: nodejs20.x
Resources:
  MyLambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: MyLambda
      Handler: app.handler
",
    ".vscode/launch.json": "{
  \\"version\\": \\"0.2.0\\",
  \\"configurations\\": [
    // BEGIN AwsSamPlugin
    {
      \\"name\\": \\"MyLambda\\",
      \\"type\\": \\"node\\",
      \\"request\\": \\"attach\\",
      \\"address\\": \\"localhost\\",
      \\"port\\": 5858,
      \\"localRoot\\": \\"\${workspaceFolder}/.aws-sam/build/MyLambda\\",
      \\"remoteRoot\\": \\"/var/task\\",
      \\"protocol\\": \\"inspector\\",
      \\"stopOnEntry\\": false,
      \\"outFiles\\": [
        \\"\${workspaceFolder}/.aws-sam/build/MyLambda/**/*.js\\"
      ],
      \\"sourceMaps\\": true,
      \\"skipFiles\\": [
        \\"/var/runtime/**/*.js\\",
        \\"<node_internals>/**/*.js\\"
      ]
    }
    // END AwsSamPlugin
  ]
}",
  },
}
`;

exports[`Happy path with multiple projects and different template names works 1`] = `
Object {
  "entryPoints": Object {
    "a#MyLambda": "./project-a/src/my-lambda/app",
    "b#MyLambda": "./project-b/src/my-lambda/app",
  },
  "files": Object {
    ".vscode/launch.json": "{
  \\"version\\": \\"0.2.0\\",
  \\"configurations\\": [
    // BEGIN AwsSamPlugin
    {
      \\"name\\": \\"a:MyLambda\\",
      \\"type\\": \\"node\\",
      \\"request\\": \\"attach\\",
      \\"address\\": \\"localhost\\",
      \\"port\\": 5858,
      \\"localRoot\\": \\"\${workspaceFolder}/project-a/.aws-sam/build/MyLambda\\",
      \\"remoteRoot\\": \\"/var/task\\",
      \\"protocol\\": \\"inspector\\",
      \\"stopOnEntry\\": false,
      \\"outFiles\\": [
        \\"\${workspaceFolder}/project-a/.aws-sam/build/MyLambda/**/*.js\\"
      ],
      \\"sourceMaps\\": true,
      \\"skipFiles\\": [
        \\"/var/runtime/**/*.js\\",
        \\"<node_internals>/**/*.js\\"
      ]
    },
    {
      \\"name\\": \\"b:MyLambda\\",
      \\"type\\": \\"node\\",
      \\"request\\": \\"attach\\",
      \\"address\\": \\"localhost\\",
      \\"port\\": 5858,
      \\"localRoot\\": \\"\${workspaceFolder}/project-b/.aws-sam/build/MyLambda\\",
      \\"remoteRoot\\": \\"/var/task\\",
      \\"protocol\\": \\"inspector\\",
      \\"stopOnEntry\\": false,
      \\"outFiles\\": [
        \\"\${workspaceFolder}/project-b/.aws-sam/build/MyLambda/**/*.js\\"
      ],
      \\"sourceMaps\\": true,
      \\"skipFiles\\": [
        \\"/var/runtime/**/*.js\\",
        \\"<node_internals>/**/*.js\\"
      ]
    }
    // END AwsSamPlugin
  ]
}",
    "project-a/.aws-sam/build/template.yaml": "AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Runtime: nodejs20.x
Resources:
  MyLambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: MyLambda
      Handler: app.handler
",
    "project-b/.aws-sam/build/template.yaml": "AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Runtime: nodejs20.x
Resources:
  MyLambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: MyLambda
      Handler: app.handler
",
  },
}
`;

exports[`Happy path with multiple projects works 1`] = `
Object {
  "entryPoints": Object {
    "a#MyLambda": "./project-a/src/my-lambda/app",
    "b#MyLambda": "./project-b/src/my-lambda/app",
  },
  "files": Object {
    ".vscode/launch.json": "{
  \\"version\\": \\"0.2.0\\",
  \\"configurations\\": [
    // BEGIN AwsSamPlugin
    {
      \\"name\\": \\"a:MyLambda\\",
      \\"type\\": \\"node\\",
      \\"request\\": \\"attach\\",
      \\"address\\": \\"localhost\\",
      \\"port\\": 5858,
      \\"localRoot\\": \\"\${workspaceFolder}/project-a/.aws-sam/build/MyLambda\\",
      \\"remoteRoot\\": \\"/var/task\\",
      \\"protocol\\": \\"inspector\\",
      \\"stopOnEntry\\": false,
      \\"outFiles\\": [
        \\"\${workspaceFolder}/project-a/.aws-sam/build/MyLambda/**/*.js\\"
      ],
      \\"sourceMaps\\": true,
      \\"skipFiles\\": [
        \\"/var/runtime/**/*.js\\",
        \\"<node_internals>/**/*.js\\"
      ]
    },
    {
      \\"name\\": \\"b:MyLambda\\",
      \\"type\\": \\"node\\",
      \\"request\\": \\"attach\\",
      \\"address\\": \\"localhost\\",
      \\"port\\": 5858,
      \\"localRoot\\": \\"\${workspaceFolder}/project-b/.aws-sam/build/MyLambda\\",
      \\"remoteRoot\\": \\"/var/task\\",
      \\"protocol\\": \\"inspector\\",
      \\"stopOnEntry\\": false,
      \\"outFiles\\": [
        \\"\${workspaceFolder}/project-b/.aws-sam/build/MyLambda/**/*.js\\"
      ],
      \\"sourceMaps\\": true,
      \\"skipFiles\\": [
        \\"/var/runtime/**/*.js\\",
        \\"<node_internals>/**/*.js\\"
      ]
    }
    // END AwsSamPlugin
  ]
}",
    "project-a/.aws-sam/build/template.yaml": "AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Runtime: nodejs20.x
Resources:
  MyLambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: MyLambda
      Handler: app.handler
",
    "project-b/.aws-sam/build/template.yaml": "AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31
Globals:
  Function:
    Runtime: nodejs20.x
Resources:
  MyLambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: MyLambda
      Handler: app.handler
",
  },
}
`;
